### ISSUE TITLE
Trying to spend only a part of coinbase UTXO always spends it all to destination address

### ISSUE THREAD

**Issue by vaklinov** [zcash/zcash#1616]
I am testing with ZCash v1.0.0-rc2: The following problem was observed twice: If I have a T address that has newly mined coins and I spend the coin base UTXO but not all of it using z_sendmany, then **two** outgoing transactions are created. One for the amount specified an another for the remainder. So the entire UTXO is sent from the T address to the destination Z address.

Example: I had a T address with 0.2155 TAZ - tmUYo3bpgovvqmxDC5jSFVKgpqttAqSQN2M. Command used to send incomplete output was:
```
zcash-cli z_sendmany tmUYo3bpgovvqmxDC5jSFVKgpqttAqSQN2M [{"address":"ztLBAPc4LS7DhnsXKSvcV4XNnFSN7PedxKsDccA3oM59yby1NycHrSDEiAPqzAnNUKcGsKU7a2wqzScQkttRRjB4aHbHbsQ","amount":0.20}]
```
Got result: [opid-4d7e481f-158a-4a49-a783-abf1e0ac8658]

Operation opid-4d7e481f-158a-4a49-a783-abf1e0ac8658 status after some time was: 
```
[{"id":"opid-4d7e481f-158a-4a49-a783-abf1e0ac8658","status":"success","creation_time":1477251523,"result":{"txid":"763924ee9ead6d77e0114aa1e2172b7a84fc75f30e458439d950079de95eb080"},"execution_secs":143.38682890}].
```

So this looks good, but the wallet shows that two (not one) transactions of type T->Z have been generated and the entire UTXO is obviously spent:
```
./zcash-cli z_listreceivedbyaddress ztLBAPc4LS7DhnsXKSvcV4XNnFSN7PedxKsDccA3oM59yby1NycHrSDEiAPqzAnNUKcGsKU7a2wqzScQkttRRjB4aHbHbsQ 0
[
    {
        "txid" : "763924ee9ead6d77e0114aa1e2172b7a84fc75f30e458439d950079de95eb080",
        "amount" : 0.20000000,
        "memo" : "f600000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000"
    },
    {
        "txid" : "763924ee9ead6d77e0114aa1e2172b7a84fc75f30e458439d950079de95eb080",
        "amount" : 0.01540000,
        "memo" : "f600000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000"
    }
]
```
In addition these two transactions have the same ID and they are not confirming fast - confirmed after 1 hour or so.

=========
So: I am not sure if all this is a feature in ZCash but I view it as a bug. I am developing the community ZCash Desktop GUI wallet: https://github.com/vaklinov/zcash-swing-wallet-ui and this behavior could be interpreted by users as a loss of ZCash! One user (T address) could try to send "some" money to another user (Z address) and find that it is all sent!

IMHO: If this is a feature, there should be some error handling in commands like z_sendmany. Alternatively the remainder of the UTXO needs to be spent to another Z address in the same wallet!

I attach my wallet for analysis if necessary:
[wallet.zip](https://github.com/zcash/zcash/files/546608/wallet.zip)

I also attach a screen-shot showing that the coins were initially mined and then two T->Z transactions are generated instead of 1: 
![uxto2](https://cloud.githubusercontent.com/assets/22178263/19629214/30ab58ae-9978-11e6-845d-7cd2cce9f7d1.png)

**Comment by daira** [comment]
This is as-designed but I strongly believe the design is wrong. If you try to send only part of a coinbase UTXO to a z-address then that should be an error; it should not silently send the change to the same z-address.

**Comment by radix42** [comment]
@daira and it is a royal pain to have to subtract the 10,000 satoshi from each amount when you are sweeping all your UTXO into a zaddr...the shell script quoting in the script I have to sweep all the mined coins in my wallet to a zaddr was...challenging

**Comment by daira** [comment]
The reason it isn't possible to spend only part of the coinbase UTXO to a z-address and send the rest as change to a t-address, is that that is intentionally disallowed by a consensus rule (#101).

**Comment by radix42** [comment]
Also, this error message when sending from a UTXO taddr to a zaddr when you forgot to subtract that 10,000 satoshi is problematit:
"Insufficient transparent funds, have 30100000, need 0 plus fee 10000"

**Comment by radix42** [comment]
Last I checked, 0+10000 < 30100000 :-)

**Comment by str4d** [comment]
@vaklinov we have fixed the `z_sendmany` error message in #1620, and added a rule to `z_sendmany` in #1629 that coinbase must be spent in entirety with no change (to make it clear that you can't partially spend coinbase). Does this satisfy your original issue?

**Comment by vaklinov** [comment]
The fixes in  #1620 and #1629 are satisfactory. Thank you.

**Comment by bugduino** [comment]
I've send a partial amount of an UTXO of mined coin from my taddr to my zaddr, the amount withdrawn from taddr was the one of the entire UXTO (as expected I suppose) but in the zaddr I only see the partial amount and ./zcash-cli z_listreceivedbyaddress list only one transaction.
I'm running version 1.0.0 on the mainnet
